examples/cli.yaml

```
@@ -0,0 +1,20 @@
+name: ion-schema-rust
+author: Ion Team
+about: An implementation of Amazon Ion Schema in Rust
+args:
+  - authorities:
+      short: a
+      long: authorities
+      required: true
+      multiple: true
+      min_values: 1
+      value_name: AUTHORITIES
+      help: Set Authorities to be added to the schema system for loading a schema
```

question—how does the syntax for authorities work? Do you specify the type of authority and then any additional args? (Eg. file system authority, and then the base file path)

---

For now there is just one authority implementation available that is the `FileSystemDocumentAuthority` hence I haven't provided additional option to specify the type of authority rather it just specifies base file path. I can change the `help` here to be more specific that this is just for file system authority.

---

> I can change the help here to be more specific that this is just for file system authority.

👍

Since this is CLI help text, I'd suggest only describing the user-facing behavior and not the library's types.

```suggestion
      help: One or more directories that will be searched for the requested schema.
```

---

> For now there is just one authority implementation available that is the FileSystemDocumentAuthority

Yes... but the Kotlin implementation also has an authority that loads `ion-schema-schemas` from a JAR. I know that Rust doesn't have JARs, but if we're going to call this argument "Authority", then we should plan on there being more than one authority.

On the other hand, I think what Zack is proposing is a really good idea. If it just describes user-facing behavior, then we don't even need to call it an Authority anywhere.